boxsdk/client/client.py

```diff
@@ -34,12 +44,37 @@ def __init__(self, oauth, network_layer=None, session=None):
             The session object to use. If None is provided then an instance of :class:`BoxSession` will be used.
         :type session:
             :class:`BoxSession`
+        :param client_id:
+            Box API key used for identifying the application the user is authenticating with.
+            Ignored if oauth instance is passed into this constructor.
+        :type client_id:
+            `unicode`
+        :param client_secret:
+            Box API secret used for making OAuth2 requests.
+            Ignored if oauth instance is passed into this constructor.
+        :type client_secret:
+            `unicode`
+        :param store_tokens:
+            Optional callback for getting access to tokens for storing them.
+            Ignored if oauth instance is passed into this constructor.
+        :type store_tokens:
+            `callable`
         """
         network_layer = network_layer or DefaultNetwork()
```

Comment:
This can be ``network_layer = network_layer or  oauth.network_layer or DefaultNetwork()`` (would need to add a ``network_layer`` property to ``OAuth2``).